x/evidence/keeper/keeper.go

```diff
@@ -25,6 +27,19 @@ type Keeper struct {
 }
 
 func NewKeeper(
```

Comment:
These API changes don't make sense to me. If we're going to keep the `x/evidence.Codec` interface, we should be explicit in having a single keeper constructor. This constructor takes a `cdc types.Codec` as it does now. The app creates the default codec and passes it here.

This is just _my_ opinion. I'd rather have an explicit argument provided instead of having the keeper by default use the default codec.